### PR TITLE
Allow insecure OIDC server for testing

### PR DESCRIPTION
**Pull request by dims**
We have a test environment where [openidmetadata.go](https://github.com/kubernetes/kubernetes/blame/master/test/images/agnhost/openidmetadata/openidmetadata.go) needs to talk to the issuer which needs a different root CA.

Since this is a conformance test, Do we mandate that the this test has to run against a secure endpoint? (and assume that the root CA in the agnhost container image is enough?)

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/assign @enj @liggitt

### CODE REVIEW

File: test/images/agnhost/openidmetadata/openidmetadata.go [hunk 1]
```diff
@@ -169,12 +170,14 @@ func gettoken() (string, error) {
 func withExternalOAuth2Client(ctx context.Context) (context.Context, error) {
 	// Use the default http transport with the system root bundle,
 	// since it's validating against the external internet.
+	client := &http.Client{
+		Transport: http.DefaultTransport,
+	}
+	client.Transport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```
Comment (liggitt) [inline test/images/agnhost/openidmetadata/openidmetadata.go:176]: disabling security when fetching the keyset doesn't seem correct

it seems like either system roots or in-cluster CA bundle should be usable for fetching the verification keys... the point of this test is supposed to be that in-cluster clients can actually use the published keyset info

**Comment by liggitt** [comment]
> Do we mandate that the this test has to run against a secure endpoint?

yes, since this uses the oidc jwt claim scheme, the issuer has to be an https URL

> and assume that the root CA in the agnhost container image is enough?

I'm less certain about the root certificates in that image being sufficient, but we also allow for including root certificates in the injected cluster root bundle used with the service account token

File: test/images/agnhost/openidmetadata/openidmetadata.go (hunk 1 above)
Comment (dims) [reply test/images/agnhost/openidmetadata/openidmetadata.go:176]: Ack, let me dig into the in-cluster CA bundle

**Comment by dims** [comment]
TIL - `including root certificates in the injected cluster root bundle` thanks! will dig into that.

/close

**Comment by liggitt** [comment]
> TIL - `including root certificates in the injected cluster root bundle` thanks! will dig into that.

you can see the code path above:

```
	log.Print("validating with in-cluster discovery")
...
	log.Print("falling back to validating with external discovery")
...
```

**Comment by dims** [comment]
Ah. i missed that. thanks @liggitt

**Pull request closed without merging**